src/Coordination/KeeperServer.cpp

```diff
@@ -442,9 +442,9 @@ void KeeperServer::shutdownRaftServer()
 
 void KeeperServer::shutdown()
 {
-    state_machine->shutdownStorage();
     state_manager->flushAndShutDownLogStore();
     shutdownRaftServer();
+    state_machine->shutdownStorage();
```

Comment:
@alesapin know any reason why we can't shutdown storage after RAFT server?
It just clears watches and ephemerals (this caused segfault).

Reply:
I don't know.